services/housekeeper/housekeeper.go

```diff
@@ -164,13 +209,54 @@ func (hk *Housekeeper) updateKnownValidators() {
 
 	for _, v := range validators {
 		pubkey := types.PubkeyHex(v.Validator.Pubkey)
-		err = hk.redis.SetKnownValidator(pubkey, v.Index)
+		err := hk.redis.SetKnownValidator(pubkey, v.Index)
 		if err != nil {
 			log.WithError(err).WithField("pubkey", pubkey).Fatal("failed to set known validator in Redis")
 		}
 	}
 }
 
+func (hk *Housekeeper) fetchValidators() map[types.PubkeyHex]beaconclient.ValidatorResponseEntry {
+	// return the first successful beacon node response
+	var result map[types.PubkeyHex]beaconclient.ValidatorResponseEntry
+	requestCtx, requestCtxCancel := context.WithCancel(context.Background())
+	defer requestCtxCancel()
+	var mu sync.Mutex
+	var wg sync.WaitGroup
+
+	for _, client := range hk.beaconClients {
```

Comment:
Since this fetch is not latency-sensitive, would it be worth it doing sequential requests? This would make the load a bit smaller for the BNs
Bonus if you keep track of which BN responded with data the last time and start with that one to avoid always waiting for a request that fails